packages/realm/src/List.ts

```diff
@@ -279,4 +279,38 @@ export class List<T = unknown> extends OrderedCollection<T> implements Partially
     }
     return result;
   }
+
+  remove(index: number) {
+    assert.inTransaction(this.realm);
+    assert.number(index, "index");
+    const { internal } = this;
+
+    //TODO I am not sure exactly if we expect a no-op for out of bounds operations or an exception
```

Comment:
You could argue that out-of-bound should be a no-op as it is more JSish. But as the methods here are not inspired by JS Array, I think we should throw if out of bound.